### ISSUE TITLE
Import board from Wekan export

### ISSUE THREAD

**Issue by xavierpriour** [wekan/wekan#436]
Now that Wekan has the ability to export its data (#396, #435), we need a way to import these.

**Comment by mquandalle** [comment]
I wasn’t able to make some progress in #430 this week-end as I wanted to, but I still this is the branch in which this development should take place. Basically we will have a full screen multi-step import, the first step being to select the import format, and then a various number of additional steps depending on the data format.

**Comment by xavierpriour** [comment]
Well, I was ready to make this import work with a minimal UI (think "import from __trello__ or __wekan__") in the new board popup. That way, it's not blocked by #430, and as the UI is the same as for the Trello import, there is no wasted dev time.

**Comment by mquandalle** [comment]
I think that it would be more efficient to avoid a “write UI as a popup” → “refactor into a different UI” path (especially given the Meteor 10-15s iteration time with UI components). What about just writing the server-side Meteor import method (that you can test simply by calling `Meteor.call('importWekanBoard', {...})` from the browser console)?

**Comment by xavierpriour** [comment]
Well, console is a pain and I'm a lazy boy :smile:, especially as we have to pass a lengthy json (Meteor roadmap long), and a member mapping, and the UI is 99% shared with the Trello component.
Mind if I put this in? I promise I'll take the UI out myself after #430 is in devel.

**Comment by mquandalle** [comment]
I just find strange to qualify “lazy” the fact of writing an UI instead of using the browser console :-p If you write a command in the console, like `Meteor.call('something` somethingElse);`, and then modify the `something` method to fix some bug, you can simply press the <kbd>↑</kbd> key to replay the last console command. I believe this is as efficient as you can get, or do I miss something?

**Comment by mquandalle** [comment]
Anyway, I’m just arguing on the principle, but go with the UI if you feel that it is a better way :)

**Comment by xavierpriour** [comment]
Thanks :smile: 
You're right in the general case. Here, the thing is I need to pass the members mapping, which the UI builds for me. And as the code is easy, I suspect I won't repeat the same call multiple times, rather try a few different scenarios, and UI beats writing the damn mappings by hand.
But again, I might be discounting the time I'll spend debugging... as usual.

**Comment by mquandalle** [comment]
(a bit off topic, but I think the real issue here is the Meteor iteration time, if we had React hot replacement writing UI would be a breeze. Switching to a stack that support this 1s iteration (React + Webpack on top of Meteor) will probably be the first thing I will experiment with just after the 0.10 release.)

**Comment by koorg** [comment]
It feels like wekan export / wekan import is meant to work at the moment.
We tried to export from a docker instance and import to a wekan dev branch, but it still seems to have an issue with user mapping, then triggs an error "Your JSON data does not include the proper information in the correct format". Is it meant to work only with Trello and NOT with Wekan at the moment ?
We miss this feature :)

**Comment by xavierpriour** [comment]
Sorry about that - let me have a look and I'll report back here!

**Comment by AndrewWillmottSI** [comment]
Also having the same issues here -- export/import doesn't work as mentioned above, you get a user remap step with all imported users represented as "0", and then "Your JSON data does not include the proper information in the correct format". I tried excising various parts of the exported data (members list, spurious board info) but no joy.

**Comment by snahelou** [comment]
Same issue here.

Could you please build the image with the latest 0.11 version please ?

Regards

**Comment by edusteinhorst** [comment]
Same issue here. Any news about this?

**Comment by snahelou** [comment]
Same issue with the latest version

**Comment by jrgifford** [comment]
Still an issue. :(

**Comment by xet7** [comment]
@GhassenRjab

Do you have time to look at this Wekan board import?

There is [related already merged pull request about Trello import](https://github.com/wekan/wekan/pulls?utf8=%E2%9C%93&q=is%3Apr%20import%20is%3Aclosed%20import)

And some existing [CSV/TSV import/export pull request](https://github.com/wekan/wekan/pull/413).

**Comment by GhassenRjab** [comment]
The export Wekan board feature already exists,
We just need to add the import feature.
To add this feature, I started with changing the "Create Board" pop-up
![image](https://user-images.githubusercontent.com/6123547/27713993-46a04c92-5d26-11e7-80b4-4e7d3b231c5f.png)
I changed "Import board from Trello" to be "Import board" that will open a new pop-up itself
![image](https://user-images.githubusercontent.com/6123547/27714083-c516ab3e-5d26-11e7-8b9d-1027ccf718bc.png)
containing these two options "From Trello" and "From Wekan"
Do I continue with this approach? or there are other suggestions?

**Comment by xet7** [comment]
@GhassenRjab 

Yes continue with this approach. Thanks!

**Comment by victort** [comment]
appreciate this effort, but I can't seem to copypasta my json into a new board. =(

**Comment by xet7** [comment]
@victort 

This is old issue. Add new issue.
